### ISSUE TITLE
Invalid model for incremental FP formula 

### ISSUE THREAD

**Issue by rainoftime** [Z3Prover/z3#4039]
Related: https://github.com/Z3Prover/z3/issues/3462, https://github.com/Z3Prover/z3/issues/3496, https://github.com/Z3Prover/z3/issues/3776

----

Hi, for the following formula,
~~~~
(set-logic QF_FPLRA)
(set-option :model_validate true)
(declare-const fpv3 Float32)
(declare-const fpv4 Float32)
(declare-const v0 Bool)
(declare-const v2 Bool)
(declare-const v1 Bool)
(declare-const v5 Bool)
(declare-const v8 Bool)
(declare-const v10 Bool)
(declare-const v13 Bool)
(declare-const v14 Bool)
(declare-const v16 Bool)
(declare-const r0 Real)
(declare-const r1 Real)
(declare-const r7 Real)
(declare-const v18 Bool)
(assert v5)
(declare-const fpv5 Float32)
(declare-const fpv6 Float32)
(push 1)
(declare-const v19 Bool)
(declare-const fpv7 Float32)
(assert (and (xor v18 (fp.gt fpv5 fpv6 ((_ to_fp 8 24) RNA 0.3777009132) ((_ to_fp 8 24) RNA 0.3777009132)) v19 v8 (fp.geq (fp.div RNA fpv3 ((_ to_fp 8 24) RTN 732517.0)) (fp.abs fpv4) ((_ to_fp 8 24) RNA 0.3777009132) (fp.abs fpv4))) v13 v0 v16 v19 (fp.lt fpv4 fpv3 ((_ to_fp 8 24) RNA 0.3777009132) ((_ to_fp 8 24) RTN 732517.0)) v14 (fp.lt (fp.abs fpv5) (fp.abs fpv5) fpv7 (fp.abs fpv5) fpv4) v13))
(push 1)
(assert (and v5 (> r7 0.5866255) (>= 0.0 r1)))
(check-sat)
(pop 1)
(assert (not (distinct v10 v13 (> r7 0.5866255))))
(check-sat)
~~~~
When executing
~~~~
z3 fname.smt2 timeout=10000
~~~~
z3 dd064a5554 gives an invalid model for the last check.
~~~~
(error "line 25 column 7: canceled")
unsat
sat
(error "line 30 column 10: an invalid model was generated")
~~~~

**Comment by zhendongsu** [comment]
See Nikolaj's comment from https://github.com/Z3Prover/z3/issues/3794: 

"there are too many invalid model for FP open at this point. They are extremely likely to be duplicates. So I will just close newbies. You can append to the older bugs if you want."

**Comment by rainoftime** [comment]
Append the test to https://github.com/Z3Prover/z3/issues/3462 or https://github.com/Z3Prover/z3/issues/3496?

**Comment by zhendongsu** [comment]
Perhaps https://github.com/Z3Prover/z3/issues/3462; thanks

**Comment by rainoftime** [comment]
Perhaps https://github.com/Z3Prover/z3/issues/3496 and https://github.com/Z3Prover/z3/issues/3496 can also be merged to https://github.com/Z3Prover/z3/issues/3462?

**Comment by rainoftime** [comment]
To https://github.com/Z3Prover/z3/issues/3462 for better debugging.

**Comment by zhendongsu** [comment]
> Perhaps #3496 and #3496 can also be merged to #3462?

You meant #3496 and #3776, and yes    @muchang @wintered
